site/DataPageV2Content.tsx

```diff
@@ -100,18 +101,59 @@ export const DataPageV2Content = ({
     )
 
     // insight link for when only a single link is shown
-    // note: this is the only place we are currently using datapageData.dataInsights,
-    // but we anticipate using it more for subsequent experiments.
     let insightsHref: string | undefined
     if (datapageData.primaryTopic) {
         const topicSlug = tagToSlugMap[datapageData.primaryTopic.topicTag]
-        if (datapageData.dataInsights?.length) {
+        if (datapageData.hasDataInsights) {
             insightsHref = `/data-insights${queryParamsToStr({ topic: datapageData.primaryTopic.topicTag })}`
         } else {
             insightsHref = `/${topicSlug}`
         }
     }
 
+    // note: assignedExperiments and isPageInExperiment should NOT be used to
+    // conditionally render content b/c it will cause a flash of content before js loads.
+    const [assignedExperiments, setAssignedExperiments] = useState<
+        Record<string, string>
+    >({})
+    const [isPageInExperiment, setIsPageInExperiment] = useState(false)
+
+    const experimentMap = useMemo(
+        () => new Map(experiments.map((exp) => [exp.id, exp])),
+        []
+    )
+
+    useEffect(() => {
+        // Only get assigned experiments on the client when cookies are available
+        if (typeof window !== "undefined") {
+            const assigned = getAssignedExperiments()
+            if (assigned) {
+                setAssignedExperiments(assigned)
+            }
+        }
+    }, [])
+
+    useEffect(() => {
+        // Calculate isPageInExperiment when assignedExperiments changes
+        if (typeof window === "undefined") {
+            return
+        }
+
+        const experimentIds = Object.keys(assignedExperiments)
+        if (experimentIds.length === 0) {
+            setIsPageInExperiment(false)
+            return
+        }
+
+        const currentPath = window.location.pathname
+        const inExperiment = experimentIds.some((expId: string) => {
+            const experiment = experimentMap.get(expId)
+            return experiment?.isUrlInPaths(currentPath) ?? false
+        })
+
+        setIsPageInExperiment(inExperiment)
+    }, [assignedExperiments, experimentMap])
+
```

Comment:
There is nothing in this section that would intrinsically be a component state, therefore the code should live outside the component.

We can have essentially the same logic without the React hooks in a helper function(s) (e.g. in the utils package, close to other experiments helpers). And then run it once on the top level (which would execute when the JS file gets loaded) and assign the result to variable(s) right below the function definition. Then we could just import that variable in this file and use it directly without worrying about React state.